src/mame/video/mrdo.cpp

```diff
@@ -185,13 +185,59 @@ void mrdo_state::mrdo_bgvideoram_w(offs_t offset, uint8_t data)
 	m_bg_tilemap->mark_tile_dirty(offset & 0x3ff);
 }
 
+/* PAL16R6CN used for protection. The game doesn't clear the screen
+   if a read from this address doesn't return the value it expects. */
+uint8_t mrdo_state::mrdo_SECRE_r()
+{
+    return m_pal_u001;
+}
+
 void mrdo_state::mrdo_fgvideoram_w(offs_t offset, uint8_t data)
 {
 	m_fgvideoram[offset] = data;
 	m_fg_tilemap->mark_tile_dirty(offset & 0x3ff);
+    
+    // protection.  each write latches a new value on IC u001 (PAL16R6)
+    uint8_t i9 = data & 0x01;
```

Comment:
These statements should be done with the BIT template expressions we have; i.e. **BIT(data, 0)** and so on.

Reply:
It has been updated.  I looked for contribution docs but it looks like the page is broken. 
https://wiki.mamedev.org/index.php/Submitting_Source_Code  
The docs/CONTRIBUTING.md is also empty.  Are there other resources to help with code submissions?

Reply:
Sadly, that page is still malfunctioning.  If you have code or data to send, use
![Submit-email](https://user-images.githubusercontent.com/9075896/149258675-cdb7211d-74c8-4dda-94fa-d24b90807508.png)